### PR TITLE
EZP-23784: Refactored preview location handling

### PR DESCRIPTION
**Pull request by bdunogier**
> [EZP-23784](https://jira.ez.no/browse/EZP-23784)

Improves the Location draft passed on to the preview template when previewing the first version of a Content. This draft does not have an id (it is set to null), but has the real parent location id and depth.

### Internal changes
Moved the location part out of `eZ\Publish\Core\Helpers\ContentPreviewHelper` to `eZ\Publish\Core\Helper\PreviewLocationProvider`. Using the API + SPI, the provider will either return the content's main location (API), or a fake location based on the content's parent location (SPI, uses eznodeassignment inside persistence).

### TODO
- [ ] Test on contentInfo status [instead of mainLocationId](https://github.com/ezsystems/ezpublish-kernel/pull/1113#discussion_r21676928)
- [ ] BC note

### CODE REVIEW

**Comment by bdunogier** [comment]
On second thought, I don't think we need to document BC for this. It improves something that was not usable, and doesn't break anything that was actually working.

File: eZ/Publish/Core/Helper/PreviewLocationProvider.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+     * @param mixed $contentInfo
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Location
+     */
+    public function loadMainLocation( $contentId )
+    {
+        $contentInfo = $this->contentService->loadContentInfo( $contentId );
+
+        // mainLocationId already exists, content has been published at least once.
+        if ( $contentInfo->mainLocationId )
```
Comment (andrerom) [inline eZ/Publish/Core/Helper/PreviewLocationProvider.php]: was this handled somewhere? :)

**Comment by andrerom** [comment]
afaik good for backport to 5.4 and 5.3 if you agree.

File: eZ/Publish/Core/Helper/PreviewLocationProvider.php (hunk 1 above)
Comment (bdunogier) [inline eZ/Publish/Core/Helper/PreviewLocationProvider.php]: Might not have been :p

Note that I like the three stages Brandon Savage described in his article about reviews: "you might", "you should", "you must". In any case, we can always handle this one in another commit, before backporting if necessary.

**Pull request merged**